Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -105,51 +124,155 @@ public void Create_PropertyDeclaredInBaseClass ()
+      Assert.That(notNullValidators.Length, Is.EqualTo(1));
+      Assert.That(notNullValidators[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WhenCalledTwice_InitializesDeferredInitializationValidationMessagesOnlyForNewlyRegisteredValidators ()
     {
+      var validationMessage1 = new DelegateBasedValidationMessage(() => "Test");
+      var validationMessage2 = new DelegateBasedValidationMessage(() => "newTest");
+
+      var validationMessageMock1 = new Mock<IValidationMessageFactory>();
+      validationMessageMock1
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage1);
+
+      var validationMessageMock2 = new Mock<IValidationMessageFactory>();
+      validationMessageMock2
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage2);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result1 = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock1.Object);
+
+      var notNullValidators1 = ((IPropertyValidationRule)result1).Validators.Cast<NotNullValidator>().ToArray();
+      Assert.That(notNullValidators1.Length, Is.EqualTo(1));
+      Assert.That(notNullValidators1[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result2 = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock2.Object);
+
+      var notNullValidators2 = ((IPropertyValidationRule)result2).Validators.Cast<NotNullValidator>().ToArray();
+      Assert.That(notNullValidators2.Length, Is.EqualTo(2));
+      Assert.That(notNullValidators2[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
+      Assert.That(notNullValidators2[1].ValidationMessage.ToString(), Is.EqualTo("newTest"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_IgnoresValidationMessagesForValidatorsWithoutADeferredInitializationValidationMessage ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      var propertyValidators = ((IPropertyValidationRule)result).Validators;
+
+      Assert.That(propertyValidators, Is.Empty);
     }
```

Comment:
This test misses part of the spec. we still need to register a validator. we simply go and do not use the `param.ValidationMessage` when instantiating the validator in the factory.